machinelearning/tensorflow.md

```diff
@@ -265,16 +272,93 @@ export TF_NEED_OPENCL_SYCL=0
 export TF_NEED_COMPUTECPP=0
 export TF_NEED_KAFKA=0
 export TF_NEED_TENSORRT=0
+
+# Configure the build setup. These are gcc build instructions,
+# so, enter "n" for Clang build option.
+# Do you want to use Clang to build TensorFlow? [Y/n]: n
 ./configure
 
 # Issue bazel build command with 'mkl_aarch64' config to enable onednn+acl backend
 bazel build --verbose_failures -s --config=mkl_aarch64  //tensorflow/tools/pip_package:build_pip_package //tensorflow:libtensorflow_cc.so //tensorflow:install_headers
 
 # Create and install the wheel
-./bazel-bin/tensorflow/tools/pip_package/build_pip_package ./wheel-TF2.12.0-py3.8-aarch64
+./bazel-bin/tensorflow/tools/pip_package/build_pip_package ./wheel-TF<version>-aarch64
+
+# The output wheel is generated in /home/ubuntu/tensorflow/wheel-TF<version>-aarch64
+pip install <wheel-TF<version>-aarch64/*.whl>
+```
+
+**Build with Clang**
 
-# The output wheel is generated in /home/ubuntu/tensorflow/wheel-TF2.12.0-py3.8-aarch64
-pip install <wheel-TF2.12.0-py3.8-aarch64/*.whl>
+```
+# Install the llvm version recommended in the TensorFlow release
+# llvm-17.0.2 is recommended for TF v2.16
+cd $HOME
+mkdir llvm_install
+cd llvm_install
+wget https://github.com/llvm/llvm-project/releases/download/llvmorg-17.0.2/clang+llvm-17.0.2-aarch64-linux-gnu.tar.xz .
+tar -xvf clang+llvm-17.0.2-aarch64-linux-gnu.tar.xz clang+llvm-17.0.2-aarch64-linux-gnu/
+sudo cp -r clang+llvm-17.0.2-aarch64-linux-gnu/* /usr
+clang --version
+
+# Install the required pip packages
+pip3 install numpy packaging
+sudo apt install patchelf
+
+# Install bazel for aarch64
+# Bazel version required depends on the TensorFlow version, please install the correct one
+# https://github.com/tensorflow/tensorflow/blob/master/.bazelversion captures the bazel version details
+# For example, tensorflow 2.16.1 build requires bazel 6.5.0
+cd $HOME
+mkdir bazel
+cd bazel
+wget https://github.com/bazelbuild/bazel/releases/download/<bazelversion>/bazel-<bazelversion>-linux-arm64
+mv bazel-<bazelversion>-linux-arm64 bazel
+chmod a+x bazel
+export PATH=/home/ubuntu/bazel/:$PATH
```

Comment:
Minor: better to install bazel into the standard /usr/local/bin so you don't need to modify PATH every time
```
BAZEL=6.5.0  # modify version as needed
wget https://github.com/bazelbuild/bazel/releases/download/$BAZEL/bazel-$BAZEL-linux-arm64 -O /usr/local/bin/bazel
chmod +x /usr/local/bin/bazel
```